lib/packetfu/protos/hsrp.rb

```diff
@@ -52,11 +52,10 @@ def self.can_parse?(str)
       return false unless UDPPacket.can_parse? str
       temp_packet = UDPPacket.new
       temp_packet.read(str)
-      if temp_packet.ip_ttl == 1 and [temp_packet.udp_sport,temp_packet.udp_dport] == [1985,1985] 
-        return true
-      else 
-        return false
-      end
+      return false unless temp_packet.eth_daddr == '01:00:5e:00:00:02' 
+      return false unless temp_packet.ip_daddr == '224.0.0.2' 
```

Comment:
The multicast address is superior to the TTL for detection of [HSRP](https://en.wikipedia.org/wiki/Hot_Standby_Router_Protocol) version 0 (aka version 1).

Admittedly, adding the `eth_daddr` check is incorrect and a bad idea.

The HSRP library in PacketFu only works for HSRP version 0. It does not work for HSRP version 2.

* The PacketFu HSRP library performs no version check.
* HSRP version 2 is not backwards compatible with HSRP version 0 [1][2].
* HSRP version 2 typically operates on port 2029 which wouldn't be parsed by this library, which checks port 1985.

The [RFC](https://tools.ietf.org/html/rfc2281) for HSRP version 0 makes reference to `224.0.0.2` in the same way it implies a `TTL` of `1`. The language is not forceful.

```
5.1  Packet formats

   The standby protocol runs on top of UDP, and uses port number 1985.
   Packets are sent to multicast address 224.0.0.2 with TTL 1.
```
HSRP version 0 supports only IPv4. It does not support IPv6. [3]

There is no RFC for HSRP version 2. It makes use of `224.0.0.102` for IPv4 and `ff02::66` for IPv6.

As for references, try the `hsrp` utility, and also refer to the following PCAPs:

* https://wiki.wireshark.org/SampleCaptures?action=AttachFile&do=get&target=hsrp.pcap
* https://wiki.wireshark.org/SampleCaptures?action=AttachFile&do=get&target=hsrp-and-ospf-in-LAN
* https://raw.githubusercontent.com/kholia/my-pcaps/master/captures/HSRP/HSRP-auth-md5-openwall.pcap
* https://raw.githubusercontent.com/kholia/my-pcaps/master/captures/HSRP/plain-HSRP.pcap
* https://github.com/kholia/my-pcaps/raw/master/HSRP/packet-captures/20-0.0-1409591337.pcap

My "very specific use case" is to detect all HSRP version 0 traffic for IPv4. [PacketFu fails and makes life harder](https://github.com/evilsocket/bettercap/pull/484). Given that HSRP version 0 operates only on IPv4, and that the PacketFu parser only works for version 0, this is not a specific use case. It's the only use case.

[1] https://netterrene.blogspot.com.au/2014/09/difference-between-hsrp-version-1-and.html
[2] https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/ipapp_fhrp/configuration/xe-3s/fhp-xe-3s-book/fhp-hsrp-v2.html
[3] https://www.cisco.com/c/en/us/support/docs/ip/hot-standby-router-protocol-hsrp/113216-ipv6-hsrp-00.html